weaveworks/eksctl
# Add resolve conflicts option on addon create

### Description



Closes https://github.com/weaveworks/eksctl-private/issues/572.

We need to support the resolve conflicts option for addon creation via the config file setting, similarly to what we already have for addon update ([reference PR](https://github.com/weaveworks/eksctl/pull/5830)). The most notable difference between the two pieces of functionality is that on creation, as opposed to update, the `preserve` value is not supported.


### Manual tests
1. Addon create using the `preserve` option.
```
Error: failed to create addon "coredns": operation error EKS: CreateAddon, https response error StatusCode: 400, RequestID: 59831a3f-8758-48d5-b743-193f5930b518, InvalidParameterException: Resolve conflicts option preserve is not supported for operation CreateAddon
```
2. Addon create using the `none` option.
```
First we'll modify the configMap of self-managed coredns addon, s.t. cache value is changed from 30 to 35.
This will result in a conflict when we run create addon.

- IAMRole: ""
  Issues:
  - Code: ConfigurationConflict
    Message: |-
      Conflicts found when trying to apply. Will not continue due to resolve conflicts mode. Conflicts:
      ConfigMap coredns - .data.Corefile
      Deployment.apps coredns - .spec.template.spec.containers[name="coredns"].image
    ResourceIDs: null
  Name: coredns
  NewerVersion: ""
  Status: CREATE_FAILED
  Version: v1.8.7-eksbuild.3
```
3. Addon create using the `overwrite` option.
```
The previously modified cache value should be overwritten to default after running addon create this time.

data:
  Corefile: |
    .:53 {
        ...
        cache 30
        ...
    }
    
Moreover, the operation shall complete successfully.

- IAMRole: ""
  Issues: null
  Name: coredns
  NewerVersion: ""
  Status: ACTIVE
  Version: v1.8.7-eksbuild.3
```

### Checklist
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
